ql/src/test/org/apache/hadoop/hive/ql/lockmgr/TestDbTxnManager2.java

```diff
@@ -4696,4 +4696,37 @@ public void testMaterializedViewEnableRewriteNonBlocking() throws Exception {
     txnMgr.rollbackTxn();
     driver.run("drop materialized view mv_tab_acid");
   }
+
+  @Test
+  public void testMaterializedViewRebuildLockForSameMV() throws Exception {
+    driver.run("drop materialized view if exists mv_tab_acid");
+    dropTable(new String[]{"tab_acid"});
+
+    driver.run("create table if not exists tab_acid (a int, b int) partitioned by (p string) " +
+        "stored as orc TBLPROPERTIES ('transactional'='true')");
+    driver.run("insert into tab_acid partition(p) (a,b,p) values(1,2,'foo'),(3,4,'bar')");
+
+    driver.run("create materialized view mv_tab_acid partitioned on (p) " +
+        "stored as orc TBLPROPERTIES ('transactional'='true') as select a, p from tab_acid where b > 1");
+
+    driver.run("insert into tab_acid partition(p) (a,b,p) values(1,2,'foo'),(3,4,'bar')");
+
+    driver.compileAndRespond("alter materialized view mv_tab_acid rebuild");
+    driver.lockAndRespond();
+
+    // rollback the transaction
+    txnMgr.rollbackTxn();
+
+    try {
+      driver.compileAndRespond("alter materialized view mv_tab_acid rebuild");
+    }
+    catch (Exception se) {
+      Assert.assertEquals(
+          "FAILED: SemanticException org.apache.hadoop.hive.ql.parse.SemanticException: Another process is rebuilding the materialized view default.mv_tab_acid",
+          se.getMessage());
+      Assert.fail("Should not throw exception here");
+    }
```

Comment:
If an exception is not expected here, than why is this Assert needed?

```
Assert.assertEquals(
          "FAILED: SemanticException org.apache.hadoop.hive.ql.parse.SemanticException: Another process is rebuilding the materialized view default.mv_tab_acid",
          se.getMessage());
```

Reply:
Before fix, it was throwing exception and after fix it shouldn't throw exception. 
If later someone changes code in such a way that again it throws exception then Assert.fail() won't give much information so added assertEquals also which will fail if it throws any different exception. I can include exception message in Assert.fail() also but anyway I feel above code is also fine.

Reply:
>  it won't give much information

AFAIK the exception message and stacktrace are present in failed test report. What is the missing information you refer to?

Reply:
I mean here I just included "Should not throw exception here" message in fail() so failed report will show only this message. Which exception is thrown and other details are will not be shown. Anyway to remove the confusion I will change Assert.fail() message and remove assertEquals().

Reply:
Looks fine now

Reply:
Yes, makes sense.